src/NetscriptFunctions.ts

```diff
@@ -288,43 +288,45 @@ export const ns: InternalAPI<NSFull> = {
     },
   growthAnalyze:
     (ctx) =>
-    (_hostname, _growth, _cores = 1) => {
-      const hostname = helpers.string(ctx, "hostname", _hostname);
-      const growth = helpers.number(ctx, "growth", _growth);
+    (_host, _multiplier, _cores = 1) => {
+      const host = helpers.string(ctx, "hostname", _host);
+      const mult = helpers.number(ctx, "multiplier", _multiplier);
       const cores = helpers.number(ctx, "cores", _cores);
 
       // Check argument validity
-      const server = helpers.getServer(ctx, hostname);
+      const server = helpers.getServer(ctx, host);
       if (!(server instanceof Server)) {
-        helpers.log(ctx, () => "Cannot be executed on this server.");
+        // Todo 2.3: Make this throw instead of returning 0?
+        helpers.log(ctx, () => `${host} is not a hackable server. Returning 0.`);
         return 0;
       }
-      if (typeof growth !== "number" || isNaN(growth) || growth < 1 || !isFinite(growth)) {
-        throw helpers.makeRuntimeErrorMsg(ctx, `Invalid argument: growth must be numeric and >= 1, is ${growth}.`);
+      if (mult < 1 || !isFinite(mult)) {
+        throw helpers.makeRuntimeErrorMsg(ctx, `Invalid argument: multiplier must be finite and >= 1, is ${mult}.`);
+      }
+      // TODO 2.3: Add assertion function for positive integer, there are a lot of places everywhere that can use this
+      if (cores < 1 || !isFinite(cores)) {
+        throw helpers.makeRuntimeErrorMsg(ctx, `Invalid argument: cores must be finite and >= 1, is ${cores}.`);
       }
 
-      return numCycleForGrowth(server, Number(growth), cores);
+      return Math.ceil(numCycleForGrowth(server, mult, cores));
     },
   growthAnalyzeCorrected:
     (ctx) =>
-    (_hostname, _targetMoney, _startMoney, _cores = 1) => {
```

Comment:
I contemplated this myself when I added the function, but I didn't do it this way because it makes the function mostly useless.

To make effective use of these functions, the server needs to already be prepped. However, a prepped server has max money, and a server with max money will always return 0 from the new version of `growthAnalyzeCorrected`.

A compromise that I just thought of is to make the single parameter `startMoney` instead. This reflects the question you actually want to ask: How many threads are needed to re-grow the server to max? If the server is not prepped, you'll get a different answer that, with skill, you can still interpolate to the question you wanted to ask.

This also makes the function useless (only) when the server has 0 money, which seems appropriate.

Reply:
I didn't change this because different server starting characteristics are exactly what formulas allows players to simulate.